### PR TITLE
Added `DataTable` and `DataTableArgument` types

### LINKED ISSUE #2926: [feature-request] export cucumber-related types
#### What are you trying to achieve?

export `DataTable` and other cucumber-related types such that we can use it in step definitions

#### What do you get instead?

I need to manually define it on our own
```ts
interface OwnTableDefinition {
  parse(): {
    raw(): string[][];
    rows(): string[][];
    rowsHash(): { [firstCol: string]: string };
    hashes(): Array<{ [colName: string]: string }>;
  }
  rows: string[][]
}
```
```ts
Given('I have products in my cart', (table: OwnTableDefinition) => {  
   table.parse()
})
```

### Details

* CodeceptJS version: 3.0.7
* NodeJS Version: 12.20.1
* Operating System: MacOS
* puppeteer || webdriverio || protractor || testcafe version (if related)

### PR DESCRIPTION
**Pull request by zaxovaiko**
## Motivation/Description of the PR
- Added `DataTable` and `DataTableArgument` types to `index.d.ts`
- Resolves #2926 

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [ ] :bug: Bug fix
- [x] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by zaxovaiko** [comment]
Hey, could you please review my changes and tell me if you need something more?

**Review by Arhell — COMMENTED** [review]
cc @DavertMik

File: lib/data/table.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
+const DataTableArgument = require("./dataTableArgument");
+
 /**
  * Datatable class to provide data driven testing
  */
 class DataTable {
-  /** @param {Array<*>} array */
+  /** @param {Array<DataTableArgument>} array */
   constructor(array) {
```
Comment (DavertMik) [inline lib/data/table.js:8]: I'm not sure this is correct
As you can actually pass any array into DataTable
DataTableArgument works only with Gherkin tables while DataTable works with classical tables as well

**Review by DavertMik — CHANGES_REQUESTED** [review]
This typings are too strict

**Comment by DavertMik** [comment]
I'm sorry it took me that long to review this PR,
I could not get a free time as due to constant missile attacks from russia I had power & internet glitches, as well as extra work on my main product Testomat.io.

**Pull request closed without merging**